frontend/src/Components/Settings/GenerateReport.tsx

```diff
@@ -0,0 +1,99 @@
+import { useState } from "react";
+import RadioComp from "../Shared/RadioComp";
+import axios from "axios";
+import { RuleAlertsResponse, rulealerts_group } from "../Alerts/AlertsTypes";
+
+async function getAllActiveAlertsInfoData(): Promise<RuleAlertsResponse> {
+    try {
+        // @ts-ignore
+        const ruleAlertsEndpoint = window._env_.REACT_APP_API_DOMAIN + "/api/getAllAlerts";
+
+        const miscongigResponse = await axios.get(ruleAlertsEndpoint,
+            { params: { rulecategory: "misconfiguration" } }
+        );
+        const misconfig_rule_alerts_group = miscongigResponse.data.rule_alerts_group.filter((group: rulealerts_group) => group.rule_data.active)
+
+        const attackResponse = await axios.get(ruleAlertsEndpoint,
+            { params: { rulecategory: "attackpath" } }
+        );
+        const attack_rule_alerts_group = attackResponse.data.rule_alerts_group.filter((group: rulealerts_group) => group.rule_data.active)
```

Comment:
can we do this all in one api call to "/api/getAllAlerts" so that we don't have to update this with new categories.

I think there should be an "all" category in the handler?